### ISSUE TITLE
Superfluous template params are reported as missing (misleading issue type/description)

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1243]
https://getpsalm.org/r/e246cdd792
```php
<?php
  /** @template-implements IteratorAggregate<int,string,object,array,float> */
  class C implements IteratorAggregate {
    public function getIterator() { yield 1 => "2"; }
  }
```
**Expected**: too many template params, expected 2, supplied 5
**Actual**:
```
Psalm output (using commit 262ba9b): 

ERROR: MissingTemplateParam - 3:22 - IteratorAggregate has unextended template params, expecting 2
```
